contracts/external/traders/Expiry.sol

```diff
@@ -154,17 +181,21 @@ contract Expiry is
         public
         onlySolo(msg.sender)
     {
-        (
-            uint256 marketId,
-            uint32 expiryTime
-        ) = parseCallArgs(data);
+        ExpiryArg[] memory expiries = abi.decode(data, (ExpiryArg[]));
 
-        // don't set expiry time for accounts with positive balance
-        if (expiryTime != 0 && !SOLO_MARGIN.getAccountPar(account, marketId).isNegative()) {
-            return;
+        for (uint256 i = 0; i < expiries.length; i++) {
+            ExpiryArg memory exp = expiries[i];
+            Require.that(
+                g_approvedSender[exp.account.owner][account.owner],
+                FILE,
+                "Unapproved sender"
+            );
+            Types.Par memory par = SOLO_MARGIN.getAccountPar(exp.account, exp.marketId);
+            uint256 expiryTime = par.isNegative()
```

Comment:
This is different behavior than the existing `Expiry` contract. Now if the account is non-negative it will reset to 0 instead of doing nothing. I feel like we should keep them the same

Reply:
I think it's better this way. Definitely not expirable, so set to 0